external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -289,6 +293,13 @@ private void doSeekRetriableTopicPartitions() {
                 kafkaConsumer.seek(rtp, acked.get(rtp).getCommittedOffset() + 1);    // Seek to last committed offset
             }
         }
+        
+        if(!retriableTopicPartitions.isEmpty()) {
+            //Pause other partitions for this poll so maxUncommittedPartitions isn't exceeded by too much
```

Comment:
I thought about it some more, and I don't think it's necessary. As long as we only have to guarantee that maxUncommittedOffsets isn't exceeded by more than maxPollRecords per partition, it should be fine to consume from all partitions. I'll remove this.